<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>BIO514 Set up</title>

<script src="site_libs/header-attrs-2.11.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="assets/mystyle.css" type="text/css" />
<link rel="stylesheet" href="assets/murdoch-fonts.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Systems Medicine</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fas fa-book-open"></span>
     
    BIO 513
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1.setup.html">Set up</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="1.1seqData.html">Sequence data</a>
    </li>
    <li>
      <a href="1.2seqProcessing.html">Sequence Processing</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fas fa-book-open"></span>
     
    BIO 514
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="2.introBIO514.html">Introduction</a>
    </li>
    <li>
      <a href="2.setup.html">Set up</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="2.1seqProcessing.html">Sequence processing</a>
    </li>
    <li>
      <a href="2.2dataCleaning.html">Data Cleaning</a>
    </li>
    <li>
      <a href="2.3dataViz.html">Data visualization</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/siobhon-egan/2022-systMed-genomics">
    <span class="fab fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BIO514 Set up</h1>

</div>


<div id="software" class="section level2">
<h2>Software</h2>
<p><strong>REQUIRED</strong> - We will be using RStudio to analyse the data set. It is recommend you have the following installed: <a href="https://rstudio.com/products/rstudio/download/"><em>RStudio version 1.4</em></a> or later and <a href="https://www.r-project.org/"><em>R version 4.0</em></a> or later. Further details on getting started in RStudio <a href="http://siobhonlegan.com/research_site/rstudio/">here</a>.</p>
<p><strong>Optional</strong> (<em>not needed for today’s workshop</em>) - We will not be doing the sequence pre-processing steps today but if you did want to do this you will need to download <a href="https://conda.io/projects/conda/en/latest/user-guide/install/index.html">conda</a> and <a href="https://qiime2.org/">QIIME2</a>.</p>
</div>
<div id="download-data" class="section level2">
<h2>Download data</h2>
<ol style="list-style-type: decimal">
<li><strong>Raw Sequence</strong> (<em>optional</em>)</li>
</ol>
<p>Raw amplicon 16S sequence data from West et al. (2020) <em>Gut</em> 69, 1452-1459. doi: <a href="http://dx.doi.org/10.1136/gutjnl-2019-319620">10.1136/gutjnl-2019-319620</a>. Download raw data from NCBI Sequence Read Archive. Project number <code>PRJNA493625</code> from <a href="https://sra-explorer.info/" class="uri">https://sra-explorer.info/</a>.</p>
<p>Rename so <code>fastq.gz</code> files in format <code>SAMPLEID_L001_R1_001.fastq.gz</code> (forward) or <code>SAMPLEID_L001_R2_001.fastq.gz</code> (reverse).</p>
<p>You will not be required to download this for today’s tutorial but if you wanted you could use this data and follow the <a href="2.1seqProcessing.html">sequence processing</a> page.</p>
<p><em>Here is a quick way to download all associated data from PRJNA493625</em></p>
<pre class="bash"><code>#!/usr/bin/env bash
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943540/SRR7943540_1.fastq.gz -o SRR7943540_16S_V4_of_human_feces_PBS61_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943540/SRR7943540_2.fastq.gz -o SRR7943540_16S_V4_of_human_feces_PBS61_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943542/SRR7943542_1.fastq.gz -o SRR7943542_16S_V4_of_human_feces_PBS5_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943542/SRR7943542_2.fastq.gz -o SRR7943542_16S_V4_of_human_feces_PBS5_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943539/SRR7943539_1.fastq.gz -o SRR7943539_16S_V4_of_human_feces_PBS43_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943539/SRR7943539_2.fastq.gz -o SRR7943539_16S_V4_of_human_feces_PBS43_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943541/SRR7943541_1.fastq.gz -o SRR7943541_16S_V4_of_human_feces_PBS68_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943541/SRR7943541_2.fastq.gz -o SRR7943541_16S_V4_of_human_feces_PBS68_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943544/SRR7943544_1.fastq.gz -o SRR7943544_16S_V4_of_human_feces_PBS63_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943544/SRR7943544_2.fastq.gz -o SRR7943544_16S_V4_of_human_feces_PBS63_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943543/SRR7943543_1.fastq.gz -o SRR7943543_16S_V4_of_human_feces_PBS39_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943543/SRR7943543_2.fastq.gz -o SRR7943543_16S_V4_of_human_feces_PBS39_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943538/SRR7943538_1.fastq.gz -o SRR7943538_16S_V4_of_human_feces_PBS49_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943538/SRR7943538_2.fastq.gz -o SRR7943538_16S_V4_of_human_feces_PBS49_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943535/SRR7943535_1.fastq.gz -o SRR7943535_16S_V4_of_human_feces_PBS30_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943535/SRR7943535_2.fastq.gz -o SRR7943535_16S_V4_of_human_feces_PBS30_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943537/SRR7943537_1.fastq.gz -o SRR7943537_16S_V4_of_human_feces_PBS50_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943537/SRR7943537_2.fastq.gz -o SRR7943537_16S_V4_of_human_feces_PBS50_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943536/SRR7943536_1.fastq.gz -o SRR7943536_16S_V4_of_human_feces_PBS29_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943536/SRR7943536_2.fastq.gz -o SRR7943536_16S_V4_of_human_feces_PBS29_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943533/SRR7943533_1.fastq.gz -o SRR7943533_16S_V4_of_human_feces_PBS44_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943533/SRR7943533_2.fastq.gz -o SRR7943533_16S_V4_of_human_feces_PBS44_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943534/SRR7943534_1.fastq.gz -o SRR7943534_16S_V4_of_human_feces_PBS67_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943534/SRR7943534_2.fastq.gz -o SRR7943534_16S_V4_of_human_feces_PBS67_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943531/SRR7943531_1.fastq.gz -o SRR7943531_16S_V4_of_human_feces_PBS26_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943531/SRR7943531_2.fastq.gz -o SRR7943531_16S_V4_of_human_feces_PBS26_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943532/SRR7943532_1.fastq.gz -o SRR7943532_16S_V4_of_human_feces_PBS25_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943532/SRR7943532_2.fastq.gz -o SRR7943532_16S_V4_of_human_feces_PBS25_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943530/SRR7943530_1.fastq.gz -o SRR7943530_16S_V4_of_human_feces_PBS27_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943530/SRR7943530_2.fastq.gz -o SRR7943530_16S_V4_of_human_feces_PBS27_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943529/SRR7943529_1.fastq.gz -o SRR7943529_16S_V4_of_human_feces_PBS28_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943529/SRR7943529_2.fastq.gz -o SRR7943529_16S_V4_of_human_feces_PBS28_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943528/SRR7943528_1.fastq.gz -o SRR7943528_16S_V4_of_human_feces_PBS21_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943528/SRR7943528_2.fastq.gz -o SRR7943528_16S_V4_of_human_feces_PBS21_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943527/SRR7943527_1.fastq.gz -o SRR7943527_16S_V4_of_human_feces_PBS22_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943527/SRR7943527_2.fastq.gz -o SRR7943527_16S_V4_of_human_feces_PBS22_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943525/SRR7943525_1.fastq.gz -o SRR7943525_16S_V4_of_human_feces_PBS24_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943525/SRR7943525_2.fastq.gz -o SRR7943525_16S_V4_of_human_feces_PBS24_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943524/SRR7943524_1.fastq.gz -o SRR7943524_16S_V4_of_human_feces_PBS62_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943524/SRR7943524_2.fastq.gz -o SRR7943524_16S_V4_of_human_feces_PBS62_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943521/SRR7943521_1.fastq.gz -o SRR7943521_16S_V4_of_human_feces_PBS47_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943521/SRR7943521_2.fastq.gz -o SRR7943521_16S_V4_of_human_feces_PBS47_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943523/SRR7943523_1.fastq.gz -o SRR7943523_16S_V4_of_human_feces_PBS42_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943523/SRR7943523_2.fastq.gz -o SRR7943523_16S_V4_of_human_feces_PBS42_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943526/SRR7943526_1.fastq.gz -o SRR7943526_16S_V4_of_human_feces_PBS23_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943526/SRR7943526_2.fastq.gz -o SRR7943526_16S_V4_of_human_feces_PBS23_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943522/SRR7943522_1.fastq.gz -o SRR7943522_16S_V4_of_human_feces_PBS41_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943522/SRR7943522_2.fastq.gz -o SRR7943522_16S_V4_of_human_feces_PBS41_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943515/SRR7943515_1.fastq.gz -o SRR7943515_16S_V4_of_human_feces_PBS6_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943515/SRR7943515_2.fastq.gz -o SRR7943515_16S_V4_of_human_feces_PBS6_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943517/SRR7943517_1.fastq.gz -o SRR7943517_16S_V4_of_human_feces_PBS46_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943517/SRR7943517_2.fastq.gz -o SRR7943517_16S_V4_of_human_feces_PBS46_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943518/SRR7943518_1.fastq.gz -o SRR7943518_16S_V4_of_human_feces_PBS45_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943518/SRR7943518_2.fastq.gz -o SRR7943518_16S_V4_of_human_feces_PBS45_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943516/SRR7943516_1.fastq.gz -o SRR7943516_16S_V4_of_human_feces_PBS38_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943516/SRR7943516_2.fastq.gz -o SRR7943516_16S_V4_of_human_feces_PBS38_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943519/SRR7943519_1.fastq.gz -o SRR7943519_16S_V4_of_human_feces_PBS48_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943519/SRR7943519_2.fastq.gz -o SRR7943519_16S_V4_of_human_feces_PBS48_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943520/SRR7943520_1.fastq.gz -o SRR7943520_16S_V4_of_human_feces_PBS57_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943520/SRR7943520_2.fastq.gz -o SRR7943520_16S_V4_of_human_feces_PBS57_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943513/SRR7943513_1.fastq.gz -o SRR7943513_16S_V4_of_human_feces_PBS8_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943513/SRR7943513_2.fastq.gz -o SRR7943513_16S_V4_of_human_feces_PBS8_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943509/SRR7943509_1.fastq.gz -o SRR7943509_16S_V4_of_human_feces_PBS4_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943509/SRR7943509_2.fastq.gz -o SRR7943509_16S_V4_of_human_feces_PBS4_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943511/SRR7943511_1.fastq.gz -o SRR7943511_16S_V4_of_human_feces_PBS2_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943511/SRR7943511_2.fastq.gz -o SRR7943511_16S_V4_of_human_feces_PBS2_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943510/SRR7943510_1.fastq.gz -o SRR7943510_16S_V4_of_human_feces_PBS1_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943510/SRR7943510_2.fastq.gz -o SRR7943510_16S_V4_of_human_feces_PBS1_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943514/SRR7943514_1.fastq.gz -o SRR7943514_16S_V4_of_human_feces_PBS34_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943514/SRR7943514_2.fastq.gz -o SRR7943514_16S_V4_of_human_feces_PBS34_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943512/SRR7943512_1.fastq.gz -o SRR7943512_16S_V4_of_human_feces_PBS33_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943512/SRR7943512_2.fastq.gz -o SRR7943512_16S_V4_of_human_feces_PBS33_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943506/SRR7943506_1.fastq.gz -o SRR7943506_16S_V4_of_human_feces_PBS32_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943506/SRR7943506_2.fastq.gz -o SRR7943506_16S_V4_of_human_feces_PBS32_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943507/SRR7943507_1.fastq.gz -o SRR7943507_16S_V4_of_human_feces_PBS10_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943507/SRR7943507_2.fastq.gz -o SRR7943507_16S_V4_of_human_feces_PBS10_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943508/SRR7943508_1.fastq.gz -o SRR7943508_16S_V4_of_human_feces_PBS3_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943508/SRR7943508_2.fastq.gz -o SRR7943508_16S_V4_of_human_feces_PBS3_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943503/SRR7943503_1.fastq.gz -o SRR7943503_16S_V4_of_human_feces_PBS59_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943503/SRR7943503_2.fastq.gz -o SRR7943503_16S_V4_of_human_feces_PBS59_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943504/SRR7943504_1.fastq.gz -o SRR7943504_16S_V4_of_human_feces_PBS60_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943504/SRR7943504_2.fastq.gz -o SRR7943504_16S_V4_of_human_feces_PBS60_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943505/SRR7943505_1.fastq.gz -o SRR7943505_16S_V4_of_human_feces_PBS9_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943505/SRR7943505_2.fastq.gz -o SRR7943505_16S_V4_of_human_feces_PBS9_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943499/SRR7943499_1.fastq.gz -o SRR7943499_16S_V4_of_human_feces_PBS54_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/009/SRR7943499/SRR7943499_2.fastq.gz -o SRR7943499_16S_V4_of_human_feces_PBS54_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943500/SRR7943500_1.fastq.gz -o SRR7943500_16S_V4_of_human_feces_PBS36_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/000/SRR7943500/SRR7943500_2.fastq.gz -o SRR7943500_16S_V4_of_human_feces_PBS36_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943502/SRR7943502_1.fastq.gz -o SRR7943502_16S_V4_of_human_feces_PBS31_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943502/SRR7943502_2.fastq.gz -o SRR7943502_16S_V4_of_human_feces_PBS31_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943497/SRR7943497_1.fastq.gz -o SRR7943497_16S_V4_of_human_feces_PBS56_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/007/SRR7943497/SRR7943497_2.fastq.gz -o SRR7943497_16S_V4_of_human_feces_PBS56_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943501/SRR7943501_1.fastq.gz -o SRR7943501_16S_V4_of_human_feces_PBS52_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943501/SRR7943501_2.fastq.gz -o SRR7943501_16S_V4_of_human_feces_PBS52_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943498/SRR7943498_1.fastq.gz -o SRR7943498_16S_V4_of_human_feces_PBS53_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/008/SRR7943498/SRR7943498_2.fastq.gz -o SRR7943498_16S_V4_of_human_feces_PBS53_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943495/SRR7943495_1.fastq.gz -o SRR7943495_16S_V4_of_human_feces_PBS58_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/005/SRR7943495/SRR7943495_2.fastq.gz -o SRR7943495_16S_V4_of_human_feces_PBS58_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943496/SRR7943496_1.fastq.gz -o SRR7943496_16S_V4_of_human_feces_PBS55_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/006/SRR7943496/SRR7943496_2.fastq.gz -o SRR7943496_16S_V4_of_human_feces_PBS55_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943492/SRR7943492_1.fastq.gz -o SRR7943492_16S_V4_of_human_feces_PBS15_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/002/SRR7943492/SRR7943492_2.fastq.gz -o SRR7943492_16S_V4_of_human_feces_PBS15_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943493/SRR7943493_1.fastq.gz -o SRR7943493_16S_V4_of_human_feces_PBS16_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/003/SRR7943493/SRR7943493_2.fastq.gz -o SRR7943493_16S_V4_of_human_feces_PBS16_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943494/SRR7943494_1.fastq.gz -o SRR7943494_16S_V4_of_human_feces_PBS37_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/004/SRR7943494/SRR7943494_2.fastq.gz -o SRR7943494_16S_V4_of_human_feces_PBS37_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943491/SRR7943491_1.fastq.gz -o SRR7943491_16S_V4_of_human_feces_PBS18_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/001/SRR7943491/SRR7943491_2.fastq.gz -o SRR7943491_16S_V4_of_human_feces_PBS18_2.fastq.gz</code></pre>
<hr />
<ol start="2" style="list-style-type: decimal">
<li><strong>Pre-processed QIIME2 files</strong> (<em>optional</em>)</li>
</ol>
<p>(<em>optional</em>)</p>
<p>Pre-processed QIIME2 sequence data as outlined in <a href="2.1seqProcessing.html">sequence processing</a> is available for download on <a href="https://figshare.com/s/0c459524e59953bf2585">FigShare link</a>.</p>
<p>You can view the downloaded <code>qzv</code> files using <a href="https://view.qiime2.org/">QIIME2 view</a>.</p>
<hr />
<ol start="3" style="list-style-type: decimal">
<li><strong>RData</strong> (<em>REQUIRED</em>)</li>
</ol>
<p>If you are following along on your own computer the easiest way is to download this GitHub repository using either option <strong>A</strong> or <strong>B</strong> below:</p>
<ol style="list-style-type: upper-alpha">
<li><p>Go to <a href="https://github.com/siobhon-egan/2022-systMed-genomics" class="uri">https://github.com/siobhon-egan/2022-systMed-genomics</a> and click on the green <strong>Code</strong> button. Select <strong>Download ZIP</strong>, open/unzip the file. Open the <code>.Rmd</code> files in RStudio you will be able to follow along for the data analysis.</p></li>
<li><p>Use terminal and clone the GitHub repo.</p></li>
</ol>
<pre><code>git clone https://github.com/siobhon-egan/2022-systMed-genomics.git</code></pre>
<blockquote>
<p>Alternatively if you are using the RStudio cloud server as part of the BIO513 unit I have include the data we will need for today’s lesson in the directory <strong>data/BIO514-microbiome</strong> directory.</p>
</blockquote>
<p>You can then just create your own Rmarkdown file and save in the <em>home/</em> directory. You can copy the code chunks in the script on the webpage <a href="http://siobhonlegan.com/2022-systMed-genomics/2.3dataViz.html">dataViz</a> or find a copy of the <code>.Rmd</code> file <a href="https://raw.githubusercontent.com/siobhon-egan/2022-systMed-genomics/master/2.3dataViz.Rmd">here</a>.</p>
</div>
<div id="install-r-packages" class="section level2">
<h2>Install R packages</h2>
<p>You should only need to do this once to install required libraries.</p>
<p>I recommend you run this bit of code line by line. (i.e. copy and paste one line of code directly in to the console and press enter).<br />
For some reason running the whole script at once doesn’t seem to be working.<br />
Also the order of installing the packages is important, please sick to this order.</p>
<p>Remember it assumes you have R version 4.0 and RStudio version 1.4.</p>
<pre class="r"><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
  install.packages(&quot;BiocManager&quot;)
# Phyloseq
BiocManager::install(&quot;phyloseq&quot;)
# Tidyverse
install.packages(&quot;tidyverse&quot;)
# Install microbiome
BiocManager::install(&quot;microbiome&quot;)
# Install microbiome utilities
install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;microsud/microbiomeutilities&quot;)
# Install DirichletMultinomial
BiocManager::install(&quot;DirichletMultinomial&quot;)
# Install MicrobiotaProcess
BiocManager::install(&quot;MicrobiotaProcess&quot;)
# Install ampvis2
install.packages(&quot;remotes&quot;)
remotes::install_github(&quot;MadsAlbertsen/ampvis2&quot;)
# Install vegan
install.packages(&quot;vegan&quot;)
# Install tidyverse
install.packages(&quot;tidyverse&quot;)
# Install ggpubr
install.packages(&quot;ggpubr&quot;)
# R color brewer
install.packages(&quot;RColorBrewer&quot;)
# Install reshape2
install.packages(&quot;reshape2&quot;)
# Install plotly
install.packages(&quot;plotly&quot;)</code></pre>
<p>Note I have not included packages that relating to “knitting” the document here. Everything should be executable within the R markdown files by running the code chunks.</p>
</div>

<hr> 
<div class="logos"><img src="assets/squaremu.png" width="50px" align="right"></div>

<br></br>

<p style="text-align: center;"><span style="color: #4F5556;"><em>Copyright, Siobhon Egan, 2022.</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
    <a href="https://github.com/siobhon-egan/BIO510-microbiome" class="fa fa-github"></a>
</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
